### PR TITLE
added the first menu for just in time

### PR DESCRIPTION
**Pull request by shejasabin**
As a GL, I want to be able to top up a client's order via USSD, so that I can adjust orders prior to distribution (PART 1)

The first menu that deals with checking if the client is a group leader and that he wants to top up for a valid account number in their group.

Link for testing: https://telerivet.com/p/0c6396c9/service/SVb47209de1fdd2cf0/edit

Steps:

Log in with the account number: 24450523
Should see option 4(top Up)-- choose it

**Enter an account number in the same group**
- 24450523 (should see a message saying how much they need to pay)
- use 26965613 for another account number in the same group

**Repeat the same process with an account number, not in the same group**
- use 16532926. should prompt the user for an account number in the same group

### CODE REVIEW

File: just-in-time/justInTime.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            var remainingLoan = 0;
+            if(client.BalanceHistory.length > 0){
+                client.latestBalanceHistory = client.BalanceHistory[0];
+                remainingLoan = client.latestBalanceHistory.TotalCredit - client.latestBalanceHistory.TotalRepayment_IncludingOverpayments;
+            }
+            if(remainingLoan > 500 ){
+                var amountToPay = remainingLoan-500;
+                global.sayText(translate('loan_payment_not_satisfied',{'$amount': amountToPay },state.vars.jitLang));
+            }
+        }
```
Comment (Bahatiroben) [inline just-in-time/justInTime.js:21]: good to have this function handle this block of functionality. would look more readable though if it was extracted and put out of registerHandlers and not exported. like in this specific file

**Comment by Bahatiroben** [comment]
> As a GL, I want to be able to top up a client's order via USSD, so that I can adjust orders prior to distribution (PART 1)
> 
> The first menu that deals with checking if the client is a group leader and that he wants to top up for a valid account number in their group.
> 
> Link for testing: https://telerivet.com/p/0c6396c9/service/SVb47209de1fdd2cf0/edit
> 
> Steps:
> 
> Log in with the account number: 24450523
> Should see option 4(top Up)-- choose it
> 
> **Enter an account number in the same group**
> 
> * 24450523 (should see a message saying how much they need to pay)
> * use 26965613 for an account number in the same group and has no debt(nothing should happen for now)
> 
> **Repeat the same process with an account number, not in the same group**
> 
> * use 16532926. should prompt the user for an account number in the same group

with the account number **16532926** from different group, I guess we expect something like  prompting the user for an account number in the **same group** yet it seems to behave in the same exact was as with account number **26965613 ** in the same group. is it the desired functionality?

File: just-in-time/justInTime.js (hunk 1 above)
Comment (shejasabin) [reply just-in-time/justInTime.js]: Yeah, it makes sense

**Comment by shejasabin** [comment]
> > As a GL, I want to be able to top up a client's order via USSD, so that I can adjust orders prior to distribution (PART 1)
> > The first menu that deals with checking if the client is a group leader and that he wants to top up for a valid account number in their group.
> > Link for testing: https://telerivet.com/p/0c6396c9/service/SVb47209de1fdd2cf0/edit
> > Steps:
> > Log in with the account number: 24450523
> > Should see option 4(top Up)-- choose it
> > **Enter an account number in the same group**
> > 
> > * 24450523 (should see a message saying how much they need to pay)
> > * use 26965613 for an account number in the same group and has no debt(nothing should happen for now)
> > 
> > **Repeat the same process with an account number, not in the same group**
> > 
> > * use 16532926. should prompt the user for an account number in the same group
> 
> with the account number **16532926** from different group, I guess we expect something like prompting the user for an account number in the **same group** yet it seems to behave in the same exact was as with account number **26965613 ** in the same group. is it the desired functionality?

Oooh, I have registered the **26965613** but forgot to enroll them to have a group. That's why it was showing the same message as the one in other group. You can check now it's showing their debt as a result of enrollment

**Review by Bahatiroben — APPROVED** [review]

**Review by nyabongo — APPROVED** [review]

**Pull request merged**